pkg/directorysync/directorysync.go

```diff
@@ -72,3 +73,13 @@ func DeleteDirectory(
 ) error {
 	return DefaultClient.DeleteDirectory(ctx, opts)
 }
+
+// PrimaryEmail is a method for finding a user's primary email (when applicable)
+func (r User) PrimaryEmail() (string, error) {
```

Comment:
Should this logic be in the client.go file, like the other methods? And then called via `DefaultClient`?

Reply:
I could be wrong here, but my gut said no. Client performs Directory Sync requests to the WorkOS API, but primary email doesn't actually make a request to the API, it just takes information that already exists from the User type and distills it.

Reply:
If I understand this correctly, I do think that the `PrimaryEmail` should be a method on the `User` object rather than being tied with the methods on the client that actually call the API.

Reply:
Ahh ok, that makes sense. It's just we only have methods that use the client that connects to the API, so that's why we have that logic in the client now.